src/vunnel/provider.py

```diff
@@ -212,3 +277,33 @@ def results_writer(self, **kwargs: Any) -> result.Writer:
             store_strategy=self.runtime_cfg.result_store,
             **kwargs,
         )
+
+
+def _fetch_listing_entry_archive(dest: str, entry: distribution.ListingEntry, logger: logging.Logger) -> str:
+    archive_path = os.path.join(dest, entry.basename())
+
+    # download the URL for the archive
+    resp = http.get(entry.url, logger=logger, stream=True)
+    resp.raise_for_status()
+    logger.debug(f"downloading {entry.url} to {archive_path}")
+    with open(archive_path, "wb") as fp:
+        for chunk in resp.iter_content(chunk_size=None):
+            fp.write(chunk)
+
+    # TODO: ensure the checksum matches whats in the listing entry
```

Comment:
@wagoodman is additional validation still needed besides `temp_ws.validate_checksums()` above? Or is this `TODO` done?

Reply:
this is a dead comment (implemented below)